src/OpenSpaceToolkit/Physics/Environment.cpp

```diff
@@ -50,17 +43,14 @@ Environment::Environment(
 )
     : instant_(anInstant),
       objects_(Array<Shared<const Object>>::Empty()),
-      centralCelestialObject_(nullptr)
+      centralCelestialObject_(aCentralCelestialObject)
 {
     objects_.reserve(anObjectArray.getSize() + 1);
 
-    const Shared<const Object> centralCelestialObject = Shared<const Object>(aCentralCelestialObject->clone());
-    centralCelestialObject_ = centralCelestialObject;
-
-    objects_.add(centralCelestialObject);
+    objects_.add(centralCelestialObject_);
     for (const auto& objectSPtr : anObjectArray)
     {
-        objects_.add(Shared<const Object>(objectSPtr->clone()));
+        objects_.add(objectSPtr);
```

Comment:
This is risky because it means you're sharing a pointer supplied from the outside, no there's no clear ownership anymore and subtle side effects could occur (what if these object points came from another Environment instance, that is still used in parallel to this one?)